connector/module.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Guewen Baconnier
+#    Copyright 2012-2013 Camptocamp SA
+#    Copyright 2015 anybox SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models
+from openerp.tools.cache import ormcache
+
+
+class IrModuleModule(models.Model):
+    _inherit = 'ir.module.module'
+
+    @ormcache(skiparg=1)
+    def is_module_installed(self, module_name):
+        return bool(len(self.env['ir.module.module'].search([
+            ('name', '=', module_name),
+            ('state', '=', 'installed')])))
+
+    def state_update(self, *args, **kwargs):
+        res = super(IrModuleModule, self).state_update(*args, **kwargs)
+        self.clear_caches()
+        return res
```

Comment:
In module_uninstall(), in addons/base/module/module.py the state is set to 'uninstalled' without calling this method. Do you want to cover this scenario? I'm guessing the previous implementation didn't so it would not be a regression.